### PR TITLE
Oveson/Refresh concurrently and clean up deprecation warnings

### PR DESCRIPTION
**Pull request by moveson**
This is parallel to the PR by @abrandoned (https://github.com/liveh2o/spectacles/pull/29). Fixed a problem with the `data: false` flag not being set. 

Pipeline is failing, but only in jRuby and only with MySQL. Looks like it may be a problem with the jdbc adapter. Will research further in the next few days.

@liveh2o

### CODE REVIEW

**Comment by moveson** [comment]
@liveh2o @abrandoned JDBC adapter for mysql2 returns results differently than the standard mysql2 adapter. I've updated the spectacles mysql2 adapter to handle this.

Pipeline is now passing.

File: lib/spectacles/schema_statements/mysql2_adapter.rb [hunk 1]
```diff
@@ -1,9 +1,47 @@
...
+
+        algorithm_string.gsub(/CREATE .*? (AS)+/i, "")
+      rescue ActiveRecord::StatementInvalid => e
+        raise "No view called #{view} found, #{e}"
+      end
+
+    private
+
+      def rows_from(result)
+        result.respond_to?(:rows) ? result.rows : result
```
Comment (moveson) [inline lib/spectacles/schema_statements/mysql2_adapter.rb]: The jdbc adapter returns an object that requires us to call `.rows`, but the standard adapter returns an object that we can map over directly. This method allows us to handle either.

File: lib/spectacles/schema_statements/postgresql_adapter.rb [hunk 2]
```diff
@@ -68,7 +68,7 @@ def materialized_view_build_query(view, name = nil)
         definition = row["definition"].strip.sub(/;$/, "")
 
         options = {}
-        options[:data] = false if ispopulated == 'f'
+        options[:data] = false if ispopulated == 'f' || ispopulated == false
```
Comment (moveson) [inline lib/spectacles/schema_statements/postgresql_adapter.rb]: Fixes a bug in which `:data => false` was not appearing in the schema using certain adapters.

File: lib/spectacles/schema_statements/sqlite3_adapter.rb [hunk 3]
```diff
@@ -21,7 +21,7 @@ def tables(name = nil, table_name = nil)
       end
 
       def generate_view_query(*columns)
-        sql = <<-SQL
```
Comment (moveson) [inline lib/spectacles/schema_statements/sqlite3_adapter.rb]: The local variable was not being used.

File: specs/adapters/postgresql_adapter_spec.rb [hunk 4]
```diff
@@ -25,44 +25,44 @@ def self.quote_column_name(name); name; end
 
   describe "#view_build_query" do
     it "should escape double-quotes returned by Postgres" do
-      test_base.view_build_query(:new_product_users).must_match(/\\"/)
+      _(test_base.view_build_query(:new_product_users)).must_match(/\\"/)
```
Comment (moveson) [inline specs/adapters/postgresql_adapter_spec.rb]: Using `_(obj)` notations eliminates minitest deprecation warnings.

File: specs/support/schema_statement_examples.rb [hunk 5]
```diff
@@ -133,55 +134,72 @@ def self.execute(query); query; end
...
 
-      it "takes a string as the view_name" do 
-        shared_base.drop_view(view_name.to_s).must_match(/#{view_name}/)
+      it "takes a string as the view_name" do
+        _(shared_base.drop_view(view_name.to_s)).must_match(/#{view_name}/)
       end
     end
   end
 
+  describe "#tables" do
```
Comment (moveson) [inline specs/support/schema_statement_examples.rb]: These new tests were helpful in diagnosing the problems arising from the incompatible results coming from the mysql jdbc adapter.

File: specs/support/view_examples.rb [hunk 6]
```diff
@@ -12,18 +10,22 @@ class NewProductUser < Spectacles::View
...
 
-        NewProductUser.duck_lovers.load.first.username.must_be @john.first_name
+      it "can have scopes" do
+        _(new_product_user.username).must_be @john.first_name
       end
 
       describe "an instance" do
         it "is readonly" do
-          NewProductUser.new.readonly?.must_be true
+          _(new_product_user.readonly?).must_be true
```
Comment (moveson) [inline specs/support/view_examples.rb]: This avoids an internal deprecation warning that resulted from calling `.new` on `NewProductUser`.

File: spectacles.gemspec [hunk 7]
```diff
@@ -22,8 +22,8 @@ Gem::Specification.new do |gem|
   # Dependencies
   #
   gem.required_ruby_version = ">= 2.0.0"
-  gem.add_dependency "activerecord", ">= 3.2.0"
-  gem.add_dependency "activesupport", ">= 3.2.0"
+  gem.add_dependency "activerecord", ">= 3.2.0", "< 6.0"
```
Comment (moveson) [inline spectacles.gemspec]: Haven't tested anything in Rails 6, so I thought this was prudent.

File: lib/spectacles/schema_statements/abstract_adapter.rb [hunk 8]
```diff
@@ -19,8 +19,8 @@ def create_view(view_name, *args)
       end
 
       def create_view_statement(view_name, create_query)
-        query = "CREATE VIEW ? AS #{create_query}"
-        query_array = [query, view_name.to_s]
+        #query = "CREATE VIEW ? AS #{create_query}"
```
[the remark this answers is not in the document]
Comment (liveh2o) [reply lib/spectacles/schema_statements/abstract_adapter.rb]: Yeah. That code seems silly.

File: lib/spectacles/schema_statements/abstract_adapter.rb [hunk 9]
```diff
@@ -32,8 +32,8 @@ def drop_view(view_name)
       end
 
       def drop_view_statement(view_name)
-        query = "DROP VIEW IF EXISTS ? "
-        query_array = [query, view_name.to_s]
+        #query = "DROP VIEW IF EXISTS ? "
```
[the remark this answers is not in the document]
Comment (liveh2o) [reply lib/spectacles/schema_statements/abstract_adapter.rb]: Yeah. That code also seems silly.

File: lib/spectacles/schema_statements/mysql2_adapter.rb (hunk 1 above)
Comment (liveh2o) [reply lib/spectacles/schema_statements/mysql2_adapter.rb]: Super weird, but you gotta do what you gotta do.

File: lib/spectacles/schema_statements/postgresql_adapter.rb (hunk 2 above)
Comment (liveh2o) [reply lib/spectacles/schema_statements/postgresql_adapter.rb]: 👍

File: lib/spectacles/schema_statements/sqlite3_adapter.rb (hunk 3 above)
Comment (liveh2o) [reply lib/spectacles/schema_statements/sqlite3_adapter.rb]: 👍

File: specs/adapters/postgresql_adapter_spec.rb (hunk 4 above)
Comment (liveh2o) [reply specs/adapters/postgresql_adapter_spec.rb]: 👍

File: specs/support/view_examples.rb (hunk 6 above)
Comment (liveh2o) [reply specs/support/view_examples.rb]: 👍

File: spectacles.gemspec (hunk 7 above)
Comment (liveh2o) [reply spectacles.gemspec]: That's reasonable. It also allows us to easily support it moving forward.

File: specs/support/schema_statement_examples.rb (hunk 5 above)
Comment (liveh2o) [reply specs/support/schema_statement_examples.rb]: 🤘

**Comment by liveh2o** [comment]
Looking good!

ezgif com-add-text

**Comment by liveh2o** [comment]
I also think this should be major version bump since we dropped support for the Mysql gem.

**Comment by moveson** [comment]
So we'll bump to gem version 2.0.0. While we're at it, since we only test Ruby 2.2 and later in CI, should we change required_ruby_version to 2.2?

File: Readme.rdoc [hunk 10]
```diff
@@ -2,7 +2,7 @@
 
 = Spectacles
 
-Spectacles adds database view functionality to ActiveRecord. It is heavily inspired by Rails SQL Views (http://github.com/aeden/rails_sql_views/) and built from the ground-up to work with Rails 3.2+.
```
Comment (moveson) [inline Readme.rdoc]: This link is dead.

**Pull request merged**